core/auth/src/main/java/software/amazon/awssdk/auth/credentials/InstanceProfileCredentialsProvider.java

```diff
@@ -80,7 +85,24 @@ public String toString() {
     }
 
     private String getToken() {
-        return EC2MetadataUtils.getToken();
```

Comment:
I wanted as part of this change to remove using `EC2MetadataUtils` at all since we want to get rid of it in the future. If you don't feel strongly about it, I'll just leave this.

Reply:
okay , If intention is getting rid of dependency on EC2MetadataUtils , then you can leave this.